pupil_src/shared_modules/task_lib/background/task.py

```diff
@@ -0,0 +1,190 @@
+"""
+(*)~---------------------------------------------------------------------------
+Pupil - eye tracking platform
+Copyright (C) 2012-2018 Pupil Labs
+
+Distributed under the terms of the GNU
+Lesser General Public License (LGPL v3.0).
+See COPYING and COPYING.LESSER for license details.
+---------------------------------------------------------------------------~(*)
+"""
+
+import abc
+import multiprocessing as mp
+from collections import namedtuple
+from ctypes import c_bool, c_double
+
+from task_lib.interface import TaskInterface
+from task_lib.background.shared_memory import SharedMemory
+
+_TaskYieldSignal = namedtuple("_TaskYieldSignal", "datum")
+_TaskCompletedSignal = namedtuple("_TaskCompletedSignal", "return_value")
+_TaskCanceledSignal = namedtuple("_TaskCanceledSignal", "")
+_TaskExceptionSignal = namedtuple("_TaskExceptionSignal", "exception traceback")
```

Comment:
I would prefer passing a tuple of strings instead of a string with spaces. It feels more explicit to me.